loader/proto.go

```diff
@@ -751,6 +723,454 @@ func (b *builder) handlePackage() (string, error) {
 	return w.String(), nil
 }
 
+func isGoType(typ string) bool {
+	switch typ {
+	case "bool",
+		"string",
+		"[]byte",
+		"float64",
+		"float32",
+		"int",
+		"int32",
+		"int64",
+		"uint32",
+		"uint64":
+		return true
+	default:
+		return false
+	}
+}
+
+func (b *builder) fromStructToAnnotation(val interface{}) string {
+	w := &strings.Builder{}
+
+	s := reflect.TypeOf(val)
+	v := reflect.ValueOf(val)
+
+	for i := 0; i < s.NumField(); i++ {
+		name := s.Field(i).Name
+		value := v.FieldByName(name)
+		if !isNilOrEmpty(value.Interface()) {
+			k := value.Kind()
+			switch {
+			case k >= reflect.Int && k <= reflect.Int64,
+				k == reflect.Float32 || k == reflect.Float64,
+				k >= reflect.Uint && k <= reflect.Uint64,
+				k == reflect.String,
+				k == reflect.Bool:
+				// The only way to know if the value
+				// is of type alias is to check its type.
+				if isGoType(value.Type().String()) {
+					b.format(w, 0, nil, "// %s: %v,\n", name, value)
+				} else {
+					pkg := strings.Split(reflect.TypeOf(value.Interface()).String(), ".")[0]
+					b.format(w, 0, nil, "// %s: %s.%v,\n", name, pkg, value)
+				}
+			case k == reflect.Map:
```

Comment:
all of these are based on `k`, so perhaps do a `switch k`? I don't imagine the first case would get more complex, as there aren't that many kinds.